src/option_learning.py

```diff
@@ -212,3 +220,206 @@ def update_segment_from_option_spec(self, segment: Segment,
+            # changing parameters is consistent with the order of the original.
+            changing_parameters = sorted(changing_parameter_set,
+                                         key=op.parameters.index)
+            del changing_parameter_set  # not used after this
+            assert changing_parameters == [
+                v for v in op.parameters if v in changing_parameters
+            ]
+
+            for segment, sub in datastore:
+                inv_sub = {v: o for o, v in sub.items()}
+                all_objects_in_operator = [inv_sub[v] for v in op.parameters]
+
+                # First, determine the absolute goal vector for this segment.
+                changing_objects = [inv_sub[v] for v in changing_parameters]
+                initial_state = segment.states[0]
+                final_state = segment.states[-1]
+                absolute_params = final_state.vec(changing_objects)
+                option_param = absolute_params - initial_state.vec(
+                    changing_objects)
+
+                # Store the option parameterization for this segment so we can
+                # use it in update_segment_from_option_spec.
+                self._segment_to_grounding[segment] = (all_objects_in_operator,
+                                                       option_param)
+                del option_param  # not used after this
+
+                # Next, create the input vectors from all object states named
+                # in the operator parameters (not just the changing ones).
+                # Note that each segment contributions multiple data points,
+                # one per action.
+                assert len(segment.states) == len(segment.actions) + 1
+                for state, action in zip(segment.states, segment.actions):
+                    state_features = state.vec(all_objects_in_operator)
+                    # Compute the relative goal vector this segment.
+                    relative_goal_vec = absolute_params - state.vec(
+                        changing_objects)
+                    # Add a bias term for regression.
+                    x = np.hstack(([1.0], state_features, relative_goal_vec))
+                    X_regressor.append(x)
+                    Y_regressor.append(action.arr)
+
+            X_arr_regressor = np.array(X_regressor, dtype=np.float32)
+            Y_arr_regressor = np.array(Y_regressor, dtype=np.float32)
+            regressor = MLPRegressor()
+            print(f"Fitting regressor with X shape: {X_arr_regressor.shape}, "
+                  f"Y shape: {Y_arr_regressor.shape}.")
+            regressor.fit(X_arr_regressor, Y_arr_regressor)
+
+            # Construct the ParameterizedOption for this operator.
+            name = f"{op.name}LearnedOption"
+            parameterized_option = _LearnedSimpleParameterizedOption(
+                name, op, regressor, changing_parameters)
+            option_specs.append((parameterized_option, list(op.parameters)))
```

Comment:
Why not just the changing_parameters here? I guess I'm still confused why exactly we want those.

Reply:
let me know if this is cleared up after I push, if not I can elaborate more

Reply:
i'm still unclear on this but let me try to explain and you tell me if i'm right:
even though the option's policy is only tasked with changing a subset of the objects' features, checking preconditions & effects might require checking other objects' features, so the option still needs to include all the operator parameters.